packages/wdio-cli/tests/launcher.test.js

```diff
@@ -345,4 +347,22 @@ describe('launcher', () => {
             expect(launcher.getRunnerId(0)).toBe('0-0')
         })
     })
+
+    describe('config options', () => {
+        it('should create directory when the config options have a outputDir option', () => {
```

Comment:
can you start launcher, mock fs-extra if needed, and spy spy on it.
This test looks for me like testing of fs-extra itself.

Am I missing something?

Reply:
I added the mock stuff to the configParser mock file? Is that not enough?

Reply:
I'm new to jest, mocking and stuff maybe I could have things wrong.

Reply:
@nami-varthakavi your test doesn't test anything wdio related .. just extend an existing test and check that `fs.ensureDirSync` was called.

Reply:
The configParserMock has the outputDir option in its so that will trigger to create the directory. Since the mocking has to be done at the constructor level i have to either create a new file just for this test and use that file and create a new launcher object.

Reply:
The configParserMock has the outputDir option in its so that will trigger to create the directory. Since the mocking has to be done at the constructor level i have to either create a new file just for this test and use that file and create a new launcher object.

Reply:
@nami-varthakavi please mock out the `fs-extra` package and just check if that package was called by the launcher.